doc/amazon_sagemaker_model_building_pipeline.rst

```diff
@@ -0,0 +1,872 @@
+
+.. code-block:: python
+
+    sklearn_processor = SKLearnProcessor(
+        framework_version="0.23-1",
+        instance_type="ml.m5.xlarge",
+        instance_count=1,
+        base_job_name="sklearn-abalone-preprocess",
+        sagemaker_session=PipelineSession(),
+        role=sagemaker.get_execution_role(),
+    )
+
+    step_process = ProcessingStep(
+        name="MyProcessingStep",
+        ...,
+        step_args = sklearn_processor.run(
+            ...,
+            outputs=[
+                ProcessingOutput(output_name="train", source="/opt/ml/processing/train"),
+            ],
+            code="./local/preprocess.py",
+            job_arguments=["--input-data", "s3://my-input"]
+        ),
+    )
+
+    step_args = estimator.fit(inputs=TrainingInput(
+        s3_data=step_process.properties.ProcessingOutputConfig.Outputs["train"].S3Output.S3Uri,
+    ))
+
+TransformStep
+````````````````
+Referable Property List:
+
+`DescribeTransformJob`_
+
+.. _DescribeTransformJob: https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_DescribeTransformJob.html#API_DescribeTransformJob_ResponseSyntax
+
+.. code-block:: python
+
+    step_transform = TransformStep(...)
+    transform_output = step_transform.TransformOutput.S3OutputPath
+
+TuningStep
+`````````````
+Referable Property List:
+
+- `DescribeHyperParameterTuningJob`_
+- `ListTrainingJobsForHyperParameterTuningJob`_
+
+.. _DescribeHyperParameterTuningJob: https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_DescribeHyperParameterTuningJob.html#API_DescribeHyperParameterTuningJob_ResponseSyntax
+.. _ListTrainingJobsForHyperParameterTuningJob: https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ListTrainingJobsForHyperParameterTuningJob.html#API_ListTrainingJobsForHyperParameterTuningJob_ResponseSyntax
+
+Example:
+
+.. code-block:: python
+
+    step_tune = TuningStep(...)
+    # tuning step can launch multiple training jobs, thus producing multiple model artifacts
+    # we can create a model with the best performance
+    best_model = Model(
+        model_data=Join(
+            on="/",
+            values=[
+                "s3://my-bucket",
+                # from DescribeHyperParameterTuningJob
+                step_tune.properties.BestTrainingJob.TrainingJobName,
+                "output/model.tar.gz",
+            ],
+    )
+    # we can also access any top-k best as we wish
+    second_best_model = Model(
+        model_data=Join(
+            on="/",
+            values=[
+                "s3://my-bucket",
+                # from ListTrainingJobsForHyperParameterTuningJob
+                step_tune.properties.TrainingJobSummaries[1].TrainingJobName,
+                "output/model.tar.gz",
+            ],
+    )
+
+The above is just an example, :code:`TuningStep` offers a helper function to generate any :code:`top-k` model data URI easily.
```

Comment:
will :code:`TuningStep` link to https://sagemaker.readthedocs.io/en/stable/workflows/pipelines/sagemaker.workflow.pipelines.html#sagemaker.workflow.steps.TuningStep ? ideally each of these steps should link to class' documentation

Reply:
using :class: for most cases now.